src/Routes/Devices/UpdateSystem.js

```diff
@@ -104,31 +134,105 @@ CurrentVersion.propTypes = {
+
+  const setUpdateEvent = (value) => {
+    setSelectedVersion(value.cells[0]);
+    setSelectedCommitID(value);
+  };
+
+  const handleUpdateEvent = async () => {
+    setIsUpdateSubmitted(true);
+    const statusMessages = {
+      onInfo: {
+        title: 'Updating system',
+        description: ` ${selectedCommitID.deviceName} was added to the queue.`,
+      },
+      onError: {
+        title: 'Error',
+        description: `An error occurred making the request`,
+      },
+    };
+
+    await apiWithToast(
+      dispatch,
+      () =>
+        updateSystem({
+          CommitID: selectedCommitID.commitID,
+          DevicesUUID: [selectedCommitID.deviceUUID],
+        }),
+      statusMessages
+    );
+
+    handleClose();
+    setIsUpdateSubmitted(false);
+  };
+
+  const handleClose = () => {
+    // Return either to the system detail, group detail, or inventory page,
+    // depending on path and from_details param
+    let destPath = paths.inventory;
+    if (match.path === paths.inventoryDetailUpdate) {
+      destPath = search.includes('from_details=true')
+        ? paths.inventoryDetail
+        : paths.inventory;
+    }
+    if (match.path === paths.fleetManagementSystemDetailUpdate) {
+      destPath = search.includes('from_details=true')
+        ? paths.fleetManagementSystemDetail
+        : paths.fleetManagementDetail;
+    }
+
+    // Construct destination path
+    const pathLen = destPath.split('/').length;
+    const dest = pathname.split('/').slice(0, pathLen).join('/');
 
-  const buildRow = (image) => ({
-    version: image?.Version,
-    release: distributionMapper[image?.Distribution],
-    additionalPackages: image?.Packages?.length || 0,
-    allPackages: image?.TotalPackages,
-    systemsRunning: image?.SystemsRunning,
-    created: (
+    history.push({ pathname: dest });
+  };
+
+  const buildRow = (image) => {
+    const version = image?.Version;
+    const release = distributionMapper[image?.Distribution];
+    const additionalPackages = image?.Packages?.length || 0;
+    const allPackages = image?.TotalPackages;
+    const systemsRunning = image?.SystemsRunning;
+    const created = (
       <span>
         <DateFormat type="relative" date={image?.CreatedAt} />
       </span>
-    ),
-    commitID: image?.CommitID,
-    deviceUUID: device?.UUID,
-    deviceName: device?.DeviceName,
-  });
-
+    );
+    const commitID = image?.CommitID;
+    const deviceUUID = device?.UUID;
+    const deviceName = device?.DeviceName;
+    return {
+      cells: [
+        version,
+        release,
+        additionalPackages,
+        allPackages,
+        systemsRunning,
+        created,
+      ],
+      commitID,
+      deviceUUID,
+      deviceName,
+      selected: selectedVersion === version,
+    };
+  };
   const currentImage = buildRow(data?.ImageInfo?.Image);
   const newImages = data?.ImageInfo?.UpdatesAvailable?.map((update) =>
     buildRow(update?.Image)
   );
 
-  return device ? (
+  return (
```

Comment:
is this not necessary anymore?
Could you explain why?

Reply:
because the General table needs to be loaded first in order to request data from the endpoint. if you look carefully, when you open the update page you will see that the table will load first with skeletons instead of data.